ifsbench/config_mixin.py

```diff
@@ -7,30 +7,39 @@
 
 from abc import ABC, abstractmethod
 from pathlib import Path
-from typing import Dict, List, Union
+from typing import Any, ClassVar, Dict, List, Type, Union
+from typing_extensions import Annotated, Literal
 
-from pydantic import BaseModel, model_validator
-from typing_extensions import Self
+from pydantic import BaseModel, model_validator, TypeAdapter, Field, computed_field
+from pydantic.fields import FieldInfo
+from pydantic_core.core_schema import ValidatorFunctionWrapHandler
 
-__all__ = ['ConfigMixin', 'PydanticConfigMixin', 'CLASSNAME', 'RESERVED_NAMES']
+
+from typing_extensions import Self, TypeAliasType
+
+__all__ = ['AbstractDataClass', 'DataClass', 'CLASSNAME', 'RESERVED_NAMES']
 
 # Reserved strings:
-# 'classname' is used in the configuration to indicate which class has to be
+# CLASSNAME is used in the configuration to indicate which class has to be
 # constructed with that configuration and cannot be used for member variables
 # in implementing classes.
-CLASSNAME = 'classname'
+CLASSNAME = 'class_name'
 RESERVED_NAMES = [
     CLASSNAME,
 ]
 
 
-class ConfigMixin(ABC):
+class DataClass(BaseModel, use_enum_values=True):
```

Comment:
I think I still prefer ConfigMixin to better describe what this is. DataClass sounds a bit too generic for me, especially since it's not just any data, it has to be something that can be converted to json/yaml. I don't really have any better ideas, will think about it some more.